### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'value', Bug?

### ISSUE THREAD

**Issue by AlexeyRaga** [confluentinc/confluent-kafka-python#108]
I am running `AvroConsumer` and it throws the following error:

```
avro-test_1  |   File "/usr/local/bin/main.py", line 39, in <module>
avro-test_1  |     msg = consumer.poll(timeout=1.0)
avro-test_1  |   File "/usr/local/lib/python3.5/dist-packages/confluent_kafka/avro/__init__.py", line 151, in poll
avro-test_1  |     if not message.value() and not message.key():
avro-test_1  | AttributeError: 'NoneType' object has no attribute 'value'
```

`/confluent_kafka/avro/__init__.py` says:

```
        message = super(AvroConsumer, self).poll(timeout)
        if not message.value() and not message.key():
            return message
        if not message.error():
            if message.value() is not None:
```

So it looks like it tries to access `.value` even before checking if the message existed.
Is it a bug?

**Comment by AlexeyRaga** [comment]
Even when topic does exist the same error is present:

```
avro-test_1  | % test-topic [0] reached end at offset 0
avro-test_1  | % test-topic [1] reached end at offset 0
avro-test_1  | % test-topic [2] reached end at offset 0
avro-test_1  | % test-topic [5] reached end at offset 0
avro-test_1  | % test-topic [4] reached end at offset 0
avro-test_1  | % test-topic [3] reached end at offset 0
avro-test_1  | % test-topic [6] reached end at offset 0
avro-test_1  | % test-topic [8] reached end at offset 0
avro-test_1  | % test-topic [17] reached end at offset 0
avro-test_1  | % test-topic [11] reached end at offset 0
avro-test_1  | % test-topic [14] reached end at offset 0
avro-test_1  | % test-topic [13] reached end at offset 0
avro-test_1  | % test-topic [7] reached end at offset 0
avro-test_1  | % test-topic [16] reached end at offset 0
avro-test_1  | % test-topic [10] reached end at offset 0
avro-test_1  | % test-topic [19] reached end at offset 0
avro-test_1  | % test-topic [9] reached end at offset 0
avro-test_1  | % test-topic [18] reached end at offset 0
avro-test_1  | % test-topic [12] reached end at offset 0
avro-test_1  | % test-topic [15] reached end at offset 0
avro-test_1  | Traceback (most recent call last):
avro-test_1  |   File "/usr/local/bin/main.py", line 39, in <module>
avro-test_1  |     msg = consumer.poll(timeout=1.0)
avro-test_1  |   File "/usr/local/lib/python3.5/dist-packages/confluent_kafka/avro/__init__.py", line 151, in poll
avro-test_1  |     if not message.value() and not message.key():
avro-test_1  | AttributeError: 'NoneType' object has no attribute 'value'
```

**Comment by AlexeyRaga** [comment]
The "normal" consumer examples do have checks for `msg == None` so it definitely looks like a bug.

**Comment by edenhill** [comment]
You are absolutely right, thanks for finding the root cause @AlexeyRaga

**Comment by edenhill** [comment]
This is addressed in PR #109
